drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -526,15 +529,37 @@ local function initialize_switch(driver, device)
     device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
   end
 
-  if num_switch_server_eps > 0 then
+  if #switch_eps > 0 and #button_eps > 0 then
+    if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
+      local dimmable_light = false
+      for _, ep in ipairs(device.endpoints) do
+        for _, dt in ipairs(ep.device_types) do
+          if dt.device_type_id == DIMMABLE_LIGHT_DEVICE_TYPE_ID then
+            dimmable_light = true
+          end
+        end
+      end
+      if dimmable_light then
+        profile_name = "light-level"
+      end
+    end
+
+    if profile_name then
+      profile_name = profile_name .. string.format("-%d-button", #button_eps)
+      device:try_update_metadata({profile = profile_name})
+      device:set_field(DEFERRED_CONFIGURE, true)
+    else
+      configure_buttons(device)
+    end
+  elseif #switch_eps > 0 then
```

Comment:
I believe this could just be
```suggestion
  elseif num_switch_server_eps > 0 then
```

And then you can remove the switch server eps check below on line 562.

Reply:
Updated as suggested in 96e319121530b6b2ad5ff7114b02531588b3bdcd!